mesos-simulation/src/main/scala/mesosphere/mesos/simulation/DriverActor.scala

```diff
@@ -103,32 +118,94 @@ class DriverActor(schedulerProps: Props) extends Actor {
       scheduler ! driver
 
     case LaunchTasks(offers, tasks) =>
-      log.debug(s"launch tasks $offers, $tasks")
-      tasks.foreach { taskInfo: TaskInfo =>
-        scheduler ! TaskStatus.newBuilder()
-          .setSource(TaskStatus.Source.SOURCE_EXECUTOR)
-          .setTaskId(taskInfo.getTaskId)
-          .setState(TaskState.TASK_STAGING)
-          .build()
+      simulateTaskLaunch(offers, tasks)
+
+    case KillTask(taskId) =>
+      log.debug(s"kill tasksToLaunch $taskId")
+
+      tasks.get(taskId.getValue) match {
+        case Some(task) =>
+          val status = TaskStatus.newBuilder()
+            .setSource(TaskStatus.Source.SOURCE_EXECUTOR)
+            .setTaskId(taskId)
+            .setState(TaskState.TASK_KILLED)
+            .build()
+          import context.dispatcher
+          context.system.scheduler.scheduleOnce(2.second, self, ChangeTaskStatus(status))
+        case None =>
+          val status = TaskStatus.newBuilder()
+            .setSource(TaskStatus.Source.SOURCE_SLAVE)
+            .setTaskId(taskId)
+            .setState(TaskState.TASK_LOST)
+            .build()
+          import context.dispatcher
+          context.system.scheduler.scheduleOnce(1.second, self, ChangeTaskStatus(status))
+      }
+
+    case ReviveOffers =>
+      scheduler ! offers
+
+    case ChangeTaskStatus(status) =>
+      changeTaskStatus(status)
+
+    case ReconcileTask(taskStatuses) =>
+      if (taskStatuses.isEmpty) {
+        tasks.values.foreach(scheduler ! _)
+      }
+      else {
+        taskStatuses.iterator.map(_.getTaskId.getValue).map(tasks).foreach(scheduler ! _)
       }
-      tasks.foreach { taskInfo: TaskInfo =>
-        scheduler ! TaskStatus.newBuilder()
+  }
+
+  def simulateTaskLaunch(offers: Seq[OfferID], tasksToLaunch: Seq[TaskInfo]): Unit = {
+    if (random.nextDouble() > 0.001) {
+      log.debug(s"launch tasksToLaunch $offers, $tasksToLaunch")
+      tasksToLaunch.foreach { taskInfo: TaskInfo =>
+        val msg = TaskStatus.newBuilder()
```

Comment:
The code inside all 3 branches is very similar. Can you extract it into a local function?